Loading diabetes dataset

In [3]:
import pandas as pd
import numpy as np

In [4]:
diabetes_df= pd.read_csv("diabetes_clean.csv")
diabetes_df.head()

pregnancies  glucose  diastolic  triceps  insulin   bmi    dpf  age  \
0            6      148         72       35        0  33.6  0.627   50   
1            1       85         66       29        0  26.6  0.351   31   
2            8      183         64        0        0  23.3  0.672   32   
3            1       89         66       23       94  28.1  0.167   21   
4            0      137         40       35      168  43.1  2.288   33   

   diabetes  
0         1  
1         0  
2         1  
3         0  
4         1

In [22]:
#  Split into training and test sets
from sklearn.model_selection import train_test_split 

X=diabetes_df[["glucose","age","bmi"]].values
y=diabetes_df["diabetes"].values

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)


In [23]:
# Import GridSearchCV , KFold
from sklearn.model_selection import GridSearchCV ,KFold

#Import Lasso
from sklearn.linear_model import Lasso

# creating kfolds
kf= KFold(n_splits=5, random_state=42, shuffle=True)

# Set up the parameter grid
param_grid = {"alpha": np.linspace(0.00001, 1, 20)}

# Instantiate lasso_cv
lasso= Lasso()
lasso_cv = GridSearchCV(lasso, param_grid, cv=kf)

# Fit to the training data
lasso_cv.fit(X_train,y_train)
print("Tuned lasso paramaters: {}".format(lasso_cv.best_params_))
print("Tuned lasso score: {}".format(lasso_cv.best_score_))

Tuned lasso paramaters: {'alpha': 0.05264105263157895}
Tuned lasso score: 0.27066290428061185


### Unfortunately, the best model only has an R-squared score of 0.27, highlighting that using the optimal hyperparameters does not guarantee a high performing model!